### ISSUE TITLE
Change-O modifications to support 10X data

### ISSUE THREAD

**Issue by ssnn-airr** [immcantation/changeo#147]
**[Original report](https://bitbucket.org/kleinstein/changeo/issues/147) by Jason Vander Heiden (Bitbucket: [javh](https://bitbucket.org/javh), GitHub: [javh](https://github.com/javh)).**

----------------------------------------

1. Add definitions for `LOCUS`, `C_CALL` and `CELL` to the standard, with mappings to the appropriate [AIRR Rearrangement](http://docs.airr-community.org/en/latest/datarep/rearrangements.html#fields) fields (`locus`, `c_call`, `cell_id`)
2. Add parsing of `LOCUS` to `change.IO.IMGTReader` and `change.IO.IHMMuneReader`. Already complete for `change.IO.IgBLASTReader`.
3. Formalize paired cloning procedure. Ideally, as a separate subcommand to DefineClones that uses the `CELL` and `FUNCTIONAL` annotations to intelligently subset sequences for VH clustering, with either pre- or post-processing of clonal assignments using VL information.
4. CreateGermlines will also need a solution for how to generate consensus germlines for cloned and paired files.

**Comment by ssnn-airr** [comment]
**Original comment by Jason Vander Heiden (Bitbucket: [javh](https://bitbucket.org/javh), GitHub: [javh](https://github.com/javh)).**

----------------------------------------

The `--10x` argument to MakeDb is overriding the IgBLAST `LOCUS` call. If we want to keep the 10X chain annotation, we should probably put it in a different field (eg, `LOCUS_10X`). Though, the 10X call use the correct nomenclature (eg, IGH) and the IgBLAST calls don't (eg, VH)...

**Comment by ssnn-airr** [comment]
**Original comment by Roy Jiang (Bitbucket: [ruoyijiangyale](https://bitbucket.org/ruoyijiangyale), ).**

----------------------------------------

1. Looks you already did that. 

I would overwrite with IgBLAST. Seems easier to implement. But 2 needs to be done before we do this. 

3/4. Steve wants to wait for more discussion on this. 

Looks like you assigned to yourself so let me know if you need anything else from me.

**Comment by ssnn-airr** [comment]
**Original comment by ssnn (Bitbucket: [ssnn](https://bitbucket.org/ssnn), GitHub: [ssnn-airr](https://github.com/ssnn-airr)).**

----------------------------------------

@{557058:4550da01-a8bf-43cd-9876-9fde3f096568} we talked about this issue in the last meeting. We think the only point left is number 2:

> Add parsing of `LOCUS` to `change.IO.IMGTReader` and `change.IO.IHMMuneReader`. Already complete for `change.IO.IgBLASTReader`

If this is correct, can we create a new issue for 2, and resolve this one?

**Comment by ssnn-airr** [comment]
**Original comment by Jason Vander Heiden (Bitbucket: [javh](https://bitbucket.org/javh), GitHub: [javh](https://github.com/javh)).**

----------------------------------------

Looks like I took care of the locus thing about a year ago \(42c0d6c\).  

If the plan is to deprecate DefineClones in favor of scoper, then we can just close this.

**Comment by ssnn-airr** [comment]
**Original comment by ssnn (Bitbucket: [ssnn](https://bitbucket.org/ssnn), GitHub: [ssnn-airr](https://github.com/ssnn-airr)).**

----------------------------------------

See also [#scoper/20](https://bitbucket.org/kleinstein/scoper/issues/20/benchmark-scoper-defineclones-prior) re deprecation of DefineClones, and #131 re “Maybe it would be better to make a MakeDb-airr command to parse any AIRR Rearrangement file and do a little standardization”. There’s [commented out code in ConvertDb](https://bitbucket.org/kleinstein/changeo/src/0b5f6d7e5e555ba7377d44945226344f9fbb10f7/bin/ConvertDb.py#lines-995:1012) to add IMGT gaps, but it needs to be tested.

**Comment by ssnn-airr** [comment]
**Original comment by Jason Vander Heiden (Bitbucket: [javh](https://bitbucket.org/javh), GitHub: [javh](https://github.com/javh)).**

----------------------------------------

Done enough.
